Makefile.PL

```diff
@@ -0,0 +1,373 @@
+                    'GD::Text'                    => 0,
+                    'Template::Plugin::GD::Image' => 0,
+                    'GD::Graph'                   => 0,
+                    GD                            => '1.20',
+                }
+            }
+        }
+    },
+    mfa => {
+        description => 'Multi-Factor Authentication',
+        prereqs     => {
+            runtime => {
+                requires => {
+                    'Auth::GoogleAuth'    => '1.01',
+                    'GD::Barcode::QRcode' => '0',
+                },
+            }
+        },
+    },
+    mod_perl => {
+        description => 'mod_perl support under Apache',
+        prereqs     => {
+            runtime => {
+                requires => {
+                    'mod_perl2'          => '1.999022',
+                    'Apache2::SizeLimit' => '0.96',
+                }
+            }
+        }
+    },
+    inbound_email => {
+        prereqs => {
+            runtime => {
+                requires => {
+                    'Email::MIME::Attachment::Stripper' => 0,
+                    'Email::Reply'                => 0,
+                }
+            }
+        },
+        description => 'Inbound Email'
+    },
+    patch_viewer => {
+        description => 'Patch Viewer',
+        prereqs => { runtime => { requires => { PatchReader => '0.9.6' } } }
+    },
+    rest => {
+        description => 'REST Interface',
+        prereqs     => {
+            runtime => {
+                requires => {
+                    'Test::Taint' => '1.06',
+                    'JSON::RPC' => 0,
+                }
+            }
+        }
+    },
+    jobqueue => {
+        description => 'Mail Queueing',
+        prereqs     => {
+            runtime => {
+                requires => { TheSchwartz => '1.10', 'Daemon::Generic' => 0 }
+            }
+        }
+    },
+);
+
+for my $file ( glob("extensions/*/Config.pm") ) {
+    my $dir  = dirname($file);
+    my $name = basename($dir);
+
+    next if -f File::Spec->catfile( $dir, "disabled" );
+    require $file;
+    my $class = "Bugzilla::Extension::$name";
+    if ( $class->can("REQUIRED_MODULES") ) {
+        foreach my $required_module ( @{ $class->REQUIRED_MODULES() } ) {
+            $requires{ $required_module->{module} }
+                = $required_module->{version};
+        }
+    }
+
+    if ( $class->can('OPTIONAL_MODULES') ) {
+        my $default_feature = 'extension_' . lc($name) . '_optional';
+        foreach my $mod ( @{ $class->OPTIONAL_MODULES } ) {
+            my @features
+                = $mod->{feature} ? @{ $mod->{feature} } : ($default_feature);
+            foreach my $feature (@features) {
+                $optional_features{$feature}{prereqs}{runtime}{requires}
+                    { $mod->{module} } = $mod->{version} // 0;
+            }
+        }
+    }
+}
+
+# BMO Customization
+my @bmo_features = grep { !/pg|oracle|mod_perl|sqlite/i } keys %optional_features;
```

Comment:
I understand why you added this but if I do --with-feature bmo I also get lots of other optional features enabled that are not used on BMO.

Change to:
+my @bmo_features = grep { !/pg|oracle|mod_perl|sqlite|auth_ldap|auth_radius|smtp_auth|updates/i } keys %optional_features;